tests/linearizability/failpoints.go

```diff
@@ -77,13 +79,15 @@ var (
 )
 
 type Failpoint interface {
-	Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster) error
+	Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster, lg *zap.Logger) error
 	Name() string
 }
 
-type killFailpoint struct{}
+type killFailpoint struct {
+	waitForSnapshot bool
+}
 
-func (f killFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster) error {
+func (f killFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster, lg *zap.Logger) error {
 	member := clus.Procs[rand.Int()%len(clus.Procs)]
```

Comment:
how about using `member := clus.Procs[(clus.WaitLeader(t)+1)%len(clus.Procs)]` to skip leader election and save time? Or do you want to test killing leader?

Reply:
The kill failpoint should not only applicable to followers. 2s wait time is fine.

Reply:
Second thought, we can definitely optimize that with your suggestion after stabilizing this test case..